src/api/app/controllers/webui/project_controller.rb

```diff
@@ -698,12 +698,16 @@ def add_maintained_project_dialog
 
   def add_maintained_project
     authorize @project, :update?
-    @project.maintained_projects.new(project: @maintained_project)
-    if @project.save
+
+    @maintained_project = params[:maintained_project]
+    maintained_project = Project.where(name: @maintained_project)
+    if maintained_project.exists?
+      @project.maintained_projects.create!(project: maintained_project.first)
```

Comment:
The reasoning is that if there is something that goes wrong, like calling create with a valid project but still getting an error, we (developers/admins of an OBS instance) want to hit an exception. So we know there is a bug.

The other case that we fixed now was just using an exception as control flow, like invalid user input. Even worse, it wasn't catched and resulted in a 404 page.